compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -496,8 +495,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             ty::Param(_) => {
                                 // Account for `fn` items like in `issue-35677.rs` to
                                 // suggest restricting its type params.
-                                let parent_body =
-                                    hir.body_owner(hir::BodyId { hir_id: self.body_id });
+                                let parent_body = hir.body_owned_by(self.body_id).hir_id;
                                 Some(hir.get(parent_body))
```

Comment:
```suggestion
                                Some(hir.get_by_def_id(self.body_id))
```

Reply:
mh maybe I'm, missing something here but we do not need to access the parent before? and then find the node?